### PR TITLE
Use `secrecy` for OpenAIConfig and AzureConfig

### PR DESCRIPTION
**Pull request by ironman5366**
Addresses #113 

`println!("{:?}", config)`:

`OpenAIConfig { api_base: "https://api.openai.com/v1", api_key: Secret([REDACTED alloc::string::String]), org_id: "" }`

You have a great project here!

### CODE REVIEW

File: async-openai/src/config.rs [hunk 1]
```diff
@@ -104,7 +95,7 @@ impl Config for OpenAIConfig {
     }
 
     fn api_key(&self) -> &str {
-        &self.api_key
+        &self.api_key.expose_secret()
```
Comment (64bit) [inline async-openai/src/config.rs:98]: This actually leaks the protection proved by `Secret<String>` in first place. 

This calls for a change to function signature to  `fn api_key() -> &Secret<String>` in `Config` trait and in both implementations.

**Comment by 64bit** [comment]
Hi @ironman5366 

Thank you for the PR, it needs a small change where we don't expose secrets without `Secret<String>` to clients. That way benefit of it propogates to all applications built using async-openai

File: async-openai/src/config.rs (hunk 1 above)
Comment (ironman5366) [reply async-openai/src/config.rs]: Ah, ofc, https://github.com/64bit/async-openai/pull/114/commits/f5dade574e42d7770f9355081aa78023873c2849

**Review by 64bit — APPROVED** [review]
Thank you. 

I'll bundle it in next release with other breaking changes, probably v0.15.0.

**Comment by TmLev** [comment]
It would be nice to re-export `secrecy::ExposeSecret` trait so that downstream users don't need to add `secrecy` crate to their applications just for that

**Comment by 64bit** [comment]
Dang! It doesn't work without adding dependency?

**Pull request merged**